File: app/src/main/java/me/hienngo/bbcsport/entity/NewsEntity.java
package me.hienngo.bbcsport.entity;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

/**
 * @author hienngo
 * @since 1/31/18
 */
@Entity(tableName = "news")
public class NewsEntity {
    @PrimaryKey(autoGenerate = true)
    private int id;
    private String news;
    private String title;
    private String description;
    private String url;
    private String urlToImage;
    private String publishedAt;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNews() {
        return news;
    }

    public void setNews(String news) {
        this.news = news;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getUrlToImage() {
        return urlToImage;
    }

    public void setUrlToImage(String urlToImage) {
        this.urlToImage = urlToImage;
    }

    public String getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(String publishedAt) {
        this.publishedAt = publishedAt;
    }
}

File: app/src/main/java/me/hienngo/bbcsport/entity/NewsEntity.kt
package me.hienngo.bbcsport.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @author hienngo
 * @since 1/31/18
 */
@Entity(tableName = "news", primaryKeys = ["title", "author", "publishedAt"])
class NewsEntity {
    var id: Int = 0
    var news: String? = null
    var title: String = ""
    var author: String = ""
    var description: String? = null
    var url: String? = null
    var urlToImage: String? = null
    var publishedAt: String = ""
    var content: String? = null
}
